include/openenclave/edger8r/common.h

```diff
@@ -0,0 +1,168 @@
+        if (oe_add_size(&total, size) != OE_OK) \
+        {                                       \
+            _result = OE_INTEGER_OVERFLOW;      \
+            goto done;                          \
+        }                                       \
+    } while (0)
+
+/**
+ * Compute and set the pointer value for the given parameter within the input
+ * buffer. Make sure that the buffer has enough space.
+ */
+#define OE_SET_IN_POINTER(argname, argsize)                                  \
+    if (pargs_in->argname)                                                   \
+    {                                                                        \
+        if ((input_buffer + input_buffer_offset + (size_t)(argsize)) >       \
+            (input_buffer + input_buffer_size))                              \
+        {                                                                    \
+            _result = OE_BUFFER_TOO_SMALL;                                   \
+            goto done;                                                       \
+        }                                                                    \
+        *(uint8_t**)&pargs_in->argname = input_buffer + input_buffer_offset; \
+        OE_ADD_SIZE(input_buffer_offset, (size_t)(argsize));                 \
+    }
+
+#define OE_SET_IN_OUT_POINTER OE_SET_IN_POINTER
+
+/**
+ * Compute and set the pointer value for the given parameter within the output
+ * buffer. Make sure that the buffer has enough space.
+ */
+#define OE_SET_OUT_POINTER(argname, argsize)                                   \
+    if (pargs_in->argname)                                                     \
+    {                                                                          \
+        if ((output_buffer + output_buffer_offset + (size_t)(argsize)) >       \
+            (output_buffer + output_buffer_size))                              \
+        {                                                                      \
+            _result = OE_BUFFER_TOO_SMALL;                                     \
+            goto done;                                                         \
+        }                                                                      \
+        *(uint8_t**)&pargs_in->argname = output_buffer + output_buffer_offset; \
+        OE_ADD_SIZE(output_buffer_offset, (size_t)(argsize));                  \
+    }
+
+/**
+ * Compute and set the pointer value for the given parameter within the output
+ * buffer. Make sure that the buffer has enough space.
+ * Also copy the contents of the corresponding in-out pointer in the input
+ * buffer.
+ */
+#define OE_COPY_AND_SET_IN_OUT_POINTER(argname, argsize)                       \
+    if (pargs_in->argname)                                                     \
+    {                                                                          \
+        if ((output_buffer + output_buffer_offset + (size_t)(argsize)) >       \
```

Comment:
Can this overflow to bypass the check? There's also a similar check in `OE_SET_IN_POINTER` and `OE_SET_OUT_POINTER`

Reply:
Good catch. I have simplified the parameter carving logic.
Can you take another pass?